test/e2e/e2e_test.go

```diff
@@ -118,9 +121,11 @@ var _ = Describe("Dash0 Operator", Ordered, ContinueOnFailure, func() {
 		By("finished AfterAll hook of test suite root")
 	})
 
-	BeforeEach(func() {
-		By("running BeforeEach hook of test suite root")
-		createDirAndDeleteOldCollectedLogs()
+	JustAfterEach(func() {
+		currentSpecReport := CurrentSpecReport()
+		if currentSpecReport.Failed() {
+			collectPodInfoAndLogs(currentSpecReport)
+		}
```

Comment:
This is the replacement for RegisterFailHandler with a custom handler. See https://onsi.github.io/ginkgo/#separating-diagnostics-collection-and-teardown-justaftereach. It's actually meant exactly for what we are doing here.